### PR TITLE
MKM 9 cards

### PR DESCRIPTION
**Pull request by Simisays**

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/melek_reforged_researcher.txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+Name:Melek, Reforged Researcher
+ManaCost:3 U R
+Types:Legendary Creature Weird Detective
+PT:*/*
+S:Mode$ Continuous | EffectZone$ All | CharacteristicDefining$ True | SetPower$ X | SetToughness$ X | Description$ CARDNAME's power and toughness are each equal to twice the number of instant and sorcery cards in your graveyard.
+S:Mode$ ReduceCost | OnlyFirstSpell$ True | Type$ Spell | ValidSpell$ Spell.Instant,Spell.Sorcery | Activator$ You | Amount$ 3 | Description$ The first instant or sorcery spell you cast each turn costs {3} less to cast.
+SVar:X:Count$ValidGraveyard Instant.YouOwn;Sorcery.YouOwn/Times.2
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/melek_reforged_researcher.txt:7]: typo

File: forge-gui/res/cardsfolder/upcoming/tunnel_tipster.txt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+Name:Tunnel Tipster
+ManaCost:1 G
+Types:Creature Mole Scout
+PT:1/1
+T:Mode$ Phase | Phase$ End of Turn | ValidPlayer$ You | CheckSVar$ FaceDown | SVarCompare$ GE1 | Execute$ TrigPutCounters | TriggerZones$ Battlefield | TriggerDescription$ At the beginning of your end step, if a face-down creature entered the battlefield under your control this turn, put a +1/+1 counter on CARDNAME.
+SVar:TrigPutCounters:DB$ PutCounter | CounterType$ P1P1 | CounterNum$ 1
+A:AB$ Mana | Cost$ T | Produced$ G | SpellDescription$ Add {G}
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/tunnel_tipster.txt:7]: .

File: forge-gui/res/cardsfolder/upcoming/escape_tunnel.txt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+Name:Escape Tunnel
+ManaCost:no cost
+Types:Land
+A:AB$ ChangeZone | Cost$ T Sac<1/CARDNAME> | Origin$ Library | Destination$ Battlefield | Tapped$ True | ChangeType$ Land.Basic | ChangeNum$ 1 | SpellDescription$ Search your library for a basic land card, put it onto the battlefield tapped, then shuffle.
+A:AB$ Effect | Cost$ T Sac<1/CARDNAME> | ValidTgts$ Creature.powerLE2 | TgtPrompt$ Select target creature with power 2 or less | RememberObjects$ Targeted | ExileOnMoved$ Battlefield | StaticAbilities$ Unblockable | AILogic$ Pump | StackDescription$ {c:Targeted} can't be blocked this turn. | SpellDescription$ Target creature with power 2 or less can't be blocked this turn.
+SVar:Unblockable:Mode$ CantBlockBy | ValidAttacker$ Card.IsRemembered | Description$ This creature can't be blocked this turn.
+DeckHas:Ability$Sacrifice
+Oracle:{T}, Sacrifice Escape Tunnel: Search your library for a basic land card, put it onto the battlefield tapped, then shuffle.\n{T}, Sacrifice Escape Tunnel: Target creature with power 2 or less can’t be blocked this turn.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/escape_tunnel.txt:8]: `’` -> `'`

File: forge-gui/res/cardsfolder/upcoming/escape_tunnel.txt [hunk 4]
```diff
@@ -0,0 +1,8 @@
+Name:Escape Tunnel
+ManaCost:no cost
+Types:Land
+A:AB$ ChangeZone | Cost$ T Sac<1/CARDNAME> | Origin$ Library | Destination$ Battlefield | Tapped$ True | ChangeType$ Land.Basic | ChangeNum$ 1 | SpellDescription$ Search your library for a basic land card, put it onto the battlefield tapped, then shuffle.
+A:AB$ Effect | Cost$ T Sac<1/CARDNAME> | ValidTgts$ Creature.powerLE2 | TgtPrompt$ Select target creature with power 2 or less | RememberObjects$ Targeted | ExileOnMoved$ Battlefield | StaticAbilities$ Unblockable | AILogic$ Pump | StackDescription$ {c:Targeted} can't be blocked this turn. | SpellDescription$ Target creature with power 2 or less can't be blocked this turn.
+SVar:Unblockable:Mode$ CantBlockBy | ValidAttacker$ Card.IsRemembered | Description$ This creature can't be blocked this turn.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/escape_tunnel.txt:6]: Why not just

File: forge-gui/res/cardsfolder/upcoming/escape_tunnel.txt [hunk 5]
```diff
@@ -0,0 +1,8 @@
+Name:Escape Tunnel
+ManaCost:no cost
+Types:Land
+A:AB$ ChangeZone | Cost$ T Sac<1/CARDNAME> | Origin$ Library | Destination$ Battlefield | Tapped$ True | ChangeType$ Land.Basic | ChangeNum$ 1 | SpellDescription$ Search your library for a basic land card, put it onto the battlefield tapped, then shuffle.
+A:AB$ Effect | Cost$ T Sac<1/CARDNAME> | ValidTgts$ Creature.powerLE2 | TgtPrompt$ Select target creature with power 2 or less | RememberObjects$ Targeted | ExileOnMoved$ Battlefield | StaticAbilities$ Unblockable | AILogic$ Pump | StackDescription$ {c:Targeted} can't be blocked this turn. | SpellDescription$ Target creature with power 2 or less can't be blocked this turn.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/escape_tunnel.txt:5]: probably need cleanup

File: forge-gui/res/cardsfolder/upcoming/homicide_investigator.txt [hunk 6]
```diff
@@ -0,0 +1,8 @@
+Name:Homicide Investigator
+ManaCost:1 B
+Types:Creature Human Detective
+PT:2/2
+T:Mode$ ChangesZoneAll | TriggerZones$ Battlefield | ValidCards$ Creature.nonToken+YouCtrl | Origin$ Battlefield | Destination$ Graveyard | Execute$ TrigInvestigate | ActivationLimit$ 1 | TriggerDescription$ Whenever one or more nontoken creatures you control die, investigate. This ability triggers only once each turn. (Create a Clue token. It's an artifact with "{2}, Sacrifice this artifact: Draw a card.”)
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/homicide_investigator.txt:5]: `”` -> `"` also in Oracle

File: forge-gui/res/cardsfolder/upcoming/homicide_investigator.txt [hunk 7]
```diff
@@ -0,0 +1,8 @@
...
+SVar:TrigInvestigate:DB$ Investigate
+DeckHas:Ability$Investigate|Token & Type$Artifact|Clue
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/homicide_investigator.txt:7]: DeckHints:Ability$Sacrifice ?

File: forge-gui/res/cardsfolder/upcoming/melek_reforged_researcher.txt [hunk 8]
```diff
@@ -0,0 +1,9 @@
+Name:Melek, Reforged Researcher
+ManaCost:3 U R
+Types:Legendary Creature Weird Detective
+PT:*/*
+S:Mode$ Continuous | EffectZone$ All | CharacteristicDefining$ True | SetPower$ X | SetToughness$ X | Description$ CARDNAME's power and toughness are each equal to twice the number of instant and sorcery cards in your graveyard.
+S:Mode$ ReduceCost | OnlyFirstSpell$ True | Type$ Spell | ValidSpell$ Spell.Instant,Spell.Sorcery | Activator$ You | Amount$ 3 | Description$ The first instant or sorcery spell you cast each turn costs {3} less to cast.
+SVar:X:Count$ValidGraveyard Instant.YouOwn,Sorcery.YouOwn/Times.2
+DeckHints:Type$Instant|Sorcery & Ability$Mill
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/melek_reforged_researcher.txt:8]: I think you can upgrade the `Type$Instant|Sorcery` to DeckNeeds and leave the Mill as a hint

File: forge-gui/res/cardsfolder/upcoming/escape_tunnel.txt (hunk 5 above)
Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/escape_tunnel.txt:5]: Doesn't the` ExileOnMoved$ Battlefield`  already take care of that ?

Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/escape_tunnel.txt]: yea that change is wrong, nothing gets remembered on the card

**Review by tool4ever — DISMISSED** [review]

File: forge-gui/res/cardsfolder/upcoming/vein_ripper.txt [hunk 9]
```diff
@@ -0,0 +1,11 @@
+Name:Vein Ripper
+ManaCost:3 B B B
+Types:Creature Vampire Assassin
+PT:6/5
+K:Flying
+K:Ward:Sac<1/Creature/creature>
+T:Mode$ ChangesZone | Origin$ Battlefield | Destination$ Graveyard | ValidCard$ Creature | TriggerZones$ Battlefield | Execute$ TrigLoseLife | TriggerDescription$ Whenever a creature dies, target opponent loses 2 life and you gain 2 life.
+SVar:TrigLoseLife:DB$ LoseLife | ValidTgts$ Opponent | TgtPrompt$ Choose an opponent to lose life | LifeAmount$ 2 | SubAbility$ DBGainLife
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/vein_ripper.txt:8]: I think just remove this `TgtPrompt` - we are targeting anyway so the default will be better

**Review by Northmoc — APPROVED** [review]
Nice work

I mainly added a whitespace line at the end of the new files 😉

**Pull request merged**